pkg/vm/vm.go

```diff
@@ -1030,9 +1034,30 @@ func (v *VM) execute(ctx *Context, op Instruction, parameter []byte) (err error)
 		v.LoadScript(script)
 
 	case RET:
-		_ = v.istack.Pop()
+		oldCtx := v.istack.Pop().Value().(*Context)
+
 		if v.istack.Len() == 0 {
```

Comment:
This `if` stays at the end in C# implementation. I think it can be possible that we return success while theirs can fail in the middle. Maybe move it down?

Reply:
ACK